### ISSUE TITLE
label alignment, text direction, multi-line support options, and a different seBindLegend

### ISSUE THREAD

**Issue by DougBurke** [gicentre/elm-vegalite#30]
I noticed a few things looking at the 4.0, 4.1, and 4.2 specs:

- add an `AlignBaseline` (or `AlignAlphabetic` since the JSON output is \"alphabetic\") - e.g. 
  [see this example](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEDGD2B2AzAlgc0gLjMSBbAhgE4DWm2kiCALgMrIBeApmQEwAMAvhwDThQAm+KvjI4AbvgA2AV0YBnMgG0cADzIBGNtygBPDZ22rWW3cY4BdHnxgJoQpXwigILqARKjIVHQAdmWL0YVKkhtSAAjfDlGSWR4fygqWB9IK1cIL2QqSQSvZNDHF0hGeDh+OPQsZ3SMqiCQqsgJGVyADRVUAEdU3hqMtSrCmopkGP4ySDVevoy5OxzPflgCOKUAVm0WNfMw+GRoBMQpaLSZxN9czul8eCosoWQxZmmZyHwVZAVGu+yE+GlJJIOEMXKdhnpBmcoCgxhM9C8+pA5lIEjglit4OtNttdvtDsdGGDEd4-BMrjcfg8ngUoW8Pl9yD8Flh-oDgTN2elORAidVXHgiKRvhcJnVgqEoJForF4hNcMh+PwFkTallmW4FUrniDiqVYOV4JVsCDavVPM1ZBN2l0eiaoANjbSYZJxgEpnbZvNUZB0fhVlhFBswFsdlA9gcyEdJCcEcMSZdrrd7ndqbH0nTPp4mX8AUC7SqihDHWcRrCAvCPVBkeq0cs-ZiA0GQ7iI1gozHK14RQFyUnhCntbT3pnvmqc2z8yDuWBeTr3ELGd3EmawlKYnFclIfAALfDhRh3aC2uNjiZrmWD9MlMoVUR2wLixoWtodboF-p3p2jF0Td206ver6-pgIG2KhpA4b4tGhJpvy8ZkomlIDjSJbDgyODZmQrJ5hysGmJCJbOq6ph4Z6KKLHWwGgcGOJhnikYEu+RTwT2iHJo8l6vGhWaniyubTq4AkzoUs7DPOWZLg+ISrlE66ygE4SwFQSS4Me6aYQpSkqSh-LXvqt4EXGZpPlIloBNab54ZMn6Ed+xHWaRVZehRGJYjR4GQQx0FMbUkm9khHE6Yi3Gjr8WH8fmVlFnyiJEXCQXDABLn1m5zZ0a2YDtjBnYsVA-nsamnYhYyvFgNhQk8lOIl8Ds1gACRzNujAEBM27KT4cgYAA9F1TyoPgAB0qBZNu0jhANyCwF1jXNfgvWMP1AC0sR1L1AAsA0AFZyAgKE2DIuDwAyLAgBwQA)

(I think this has been in Vega-Lite since at least 3.4.0)

- add a `TextDirection` type that supports "ltr" and "rtl" (see `"dir"` field) along with support for the `"ellipsis"` and `"limit"` fields for text: https://vega.github.io/vega-lite/docs/text.html 

- '"lineBreak"' and '"lineHeight"' properties for (some) marks, although how linebreak works with your "auto-split on \n" support is an open question. I'd link to a page describing them but they only seem to be in the v4.x schema.

- for the `seBindLegend` selection property, I ended up going a slightly different way than you: rather than taking a list of `BindLegendProperty` I made it take a single `BindLegendProperty` with the following structure, which simplifies the processing a bit (at least, it makes it clear that you must specify either a `FieldName` (which is just a synomym for String/Text) or a `Channel`

```
data BindLegendProperty
  = BLField FieldName
    -- ^ The data field which should be made interactive in the legend
    --   on a single click.
  | BLChannel Channel
    -- ^ Which channel should be made interactive in a legend
    --   on a single click.
  | BLFieldEvent FieldName T.Text
    -- ^ The data field which should be made interactive in the legend and the
    --   <https://vega.github.io/vega/docs/event-streams Vega event stream>
    --   that should trigger the selection.
  | BLChannelEvent Channel T.Text
    -- ^ Which channel should be made interactive in a legend and the
    --   <https://vega.github.io/vega/docs/event-streams Vega event stream>
    --   that should trigger the selection.
```

The alternative I came up with was two constructors `BLField` and `BLChannel` (for sake of discussion) whose second argument is `Maybe Text`, but I think I prefer the slightly-more explicit version here.

**Comment by jwoLondon** [comment]
Thanks again for spotting the missing elements from the schema. I've added the alphabetic baseline, text direction and line height options.

I note that text mark literals (via `tStr`) allow for multi-line strings to be respected when providing a JSON array of strings, but not for [encoded text data](https://github.com/gicentre/elm-vegalite/blob/74a49d4ba7c51ce36c2cacbe2c9259f869ce2896/tests/src/TextFormatTests.elm#L152). Not sure if this is a Vega-Lite bug or feature, but for now I am not converting `\n` in encoded data into string arrays.

To avoid over-inflation of the API surface area, I currently do not intend to implement `lineBreak` as the `\n` seems to cover all the cases I can foresee (we could always transform input data with regex, find/replace etc. if we had to). If you think there is a compelling case for `lineBreak`, happy to add it though.
